### ISSUE TITLE
Error fatal: 'string' file not found

### ISSUE THREAD

**Issue by boodoolak** [mono/CppSharp#830]
##### Brief Description
Hi,
I have this error. How can I fix it? C++# from Nuget.
```
Parsing libraries...
Parsed 'hhive.lib'
Parsing code...
Error parsing 'ive.h'
C:\dst\include\ive.h(4,10): fatal: 'string' file not found
```

OS: Windows / OS X / Linux (include version and/or distro)
W10 AU x64

##### Used headers
```
#ifndef IVE_H
#define IVE_H

#include <string>     <--this is line 4
#include <functional>
#include <map>
#include <unordered_map>

namespace ive {

using namespace std;
...
```

##### Used settings
```
            var options = driver.Options;
            options.GeneratorKind = GeneratorKind.CSharp;
            var module = options.AddModule("dst");
            module.IncludeDirs.Add("C:\\dst\\include");
            module.Headers.Add("ive.h");
            module.LibraryDirs.Add("C:\\dst\\lib");
            module.Libraries.Add("ive.lib");
```

Target: MSVC/GCC/Clang

Other settings

**Comment by ddobrev** [comment]
We were able to solve a similar problem with the Android NDK today. Could you please specify what system headers you use?

**Comment by boodoolak** [comment]
@ddobrev 
Only these. You say about them?
```
#include <string>
#include <functional>
#include <map>
#include <unordered_map>
```

**Comment by ddobrev** [comment]
@boodoolak I am sorry, I wasn't exact in my question. What I meant was: what environment provides your system headers? Is it MSVC, MinGW, Android NDK, sth else?

**Comment by boodoolak** [comment]
@ddobrev MSVS.
The problem was solved just now after installing VS2017 and all components for c++.
I'm sorry.

But now there is another problem:
```
Error parsing 'hhive.h'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\include\xstring(1905,26): error: constexpr variable '_Memcpy_move_offset' must be initialized by a constant expression
```

Unfortunately I don't know c++. What does this error mean? How can I fix it?

**Comment by tritao** [comment]
It's probably related due to us using an "outdated" Clang revision that does not play nice with some of VS2017 headers. Using VS2015 or older for parsing is probably the most compatible right now.

**Comment by tritao** [comment]
By the way the initial issue could be caused by not setting up MSVC headers.

See `SetupMSVC` method in https://github.com/mono/CppSharp/blob/a954c54a6cd2f84dedf987c2812f48ca2a8e8ee1/src/Parser/ParserOptions.cs#L17 .

**Comment by boodoolak** [comment]
After installing vs2017 old vs2015 began using the path from vs2017 (C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC...).
I'll try to remove vs2017  and then reinstall vs2015. 
Most likely the problem is in the installation vs2015.

**Comment by boodoolak** [comment]
Yes, removing vs2017 is helped, Library was generated successfully.
That means this string?
`The query API is not registered. Assuming no instances are installed.`
![image](https://cloud.githubusercontent.com/assets/3179176/26103570/cf9bf71e-3a42-11e7-9b77-e903a84e3edb.png)

**Comment by ddobrev** [comment]
Yes, this output means success. If you use the option for automatic compilation, your binding will also be compiled for you.

**Comment by boodoolak** [comment]
@ddobrev @tritao Thank you!

**Comment by boodoolak** [comment]
I added 
`options.CompileCode = true;` 
but conpile failed
`Compiling HiveLibrary...
C:\dst\Output\HiveLibrary.cs(243,33) : error CS0234: The type or namespace name 'BasicString' does not exist in the namespace 'Std' (are you missing an assembly reference?)`
